src/views/modals/ExampleFormModal.tsx

```diff
@@ -0,0 +1,130 @@
+                </div>
+                <div className="modal-footer modal-footer_stack">
+                    <button onClick={this.props.closeModal}>
+                        {'Cancel'}
+                    </button>
+                    <button onClick={this._onClickAcceptHandler}>
+                        {'Accept'}
+                    </button>
+                </div>
+            </section>
+        );
+    }
+
+    private _buildVaporFormJsx(): JSX.Element {
+        return (
+            <form
+                autoComplete="off"
+                className="modalForm u-validate"
+                noValidate={true}
+                ref={(element: HTMLFormElement) => {
+                    this._formElement = element;
+                }}
+            >
+                <p className="modalForm-warning">
+                    {'This is an example of a custom modal. The "isRequired" attribute is set to "true" which prevents the user from clicking the esc key or click outside of the modal to close it. It also has form validation.' }
+                </p>
+                <div className="modalForm-item">
+                    <label
+                        htmlFor="vapor-modal-form-name"
+                        className="modalForm-item-label"
+                    >
+                        {'Name'}<sup>{'*'}</sup>
+                    </label>
+                    <div className="modalForm-item-input">
+                        <InputField
+                            id={'vapor-modal-form-name'}
+                            name="name"
+                            isRequired={true}
+                        />
+                    </div>
+                </div>
+                <div className="modalForm-item">
+                    <label
+                        htmlFor="vapor-modal-form-exposure-limit"
+                        className="modalForm-item-label"
+                    >
+                        {'Age'}<sup>{'*'}</sup>
+                    </label>
+                    <div className="modalForm-item-input">
+                        <InputField
+                            id={'example-form-age'}
+                            name="age"
+                            step={'1'}
+                            type={'number'}
+                            isRequired={true}
+                        />
+                    </div>
+                </div>
+                <div className="modalForm-item">
+                    <label
+                        htmlFor="vapor-modal-form-molecular-weight"
+                        className="modalForm-item-label"
+                    >
+                        {'Email'}<sup>{'*'}</sup>
+                    </label>
+                    <div className="modalForm-item-input">
+                        <InputField
+                            id={'example-form-modal-email'}
+                            name="email"
+                            pattern="[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,3}$"
```

Comment:
you could store regexes in a central constants file and then reference them here.

Reply:
For this example I just trying to keep things simple. Not to many extra files.